services/credential-server/src/apis/credential.api.ts

```diff
@@ -36,12 +37,21 @@ async function requestDisclosure(req: Request, res: Response): Promise<void> {
 
 async function revokeCredential(req: Request, res: Response): Promise<void> {
   const { credentialId, holder } = req.body;
-  await Agent.agent.revokeCredential(credentialId, holder);
-  const response: ResponseData<string> = {
+  let response: ResponseData<string> = {
     statusCode: 200,
     success: true,
     data: "Revoke credential successfully",
   };
+  try {
+    await Agent.agent.revokeCredential(credentialId, holder);
+  } catch (error) {
+    log({ error: (error as Error).message })
+    response = {
+      statusCode: 500,
```

Comment:
this should be 409 if its already revoked, or 404 if not found. Otherwise it can be 500